command/agent/config/config.go

```diff
@@ -169,14 +171,253 @@ func NewConfig() *Config {
 	}
 }
 
+// Merge merges two Agent configurations.
+func (c *Config) Merge(c2 *Config) *Config {
+	if c2 == nil {
+		return c
+	}
+
+	result := NewConfig()
+
+	result.SharedConfig = c.SharedConfig
+	if c2.SharedConfig != nil {
+		result.SharedConfig = c.SharedConfig.Merge(c2.SharedConfig)
+	}
+
+	result.AutoAuth = c.AutoAuth
+	if c2.AutoAuth != nil {
+		result.AutoAuth = c2.AutoAuth
+	}
+
+	result.Cache = c.Cache
```

Comment:
What if use_auto_auth_token is set in one file and persist in the other?

Reply:
There is only one `cache` block possible. We use the same strategy as we do for Vault itself, where we take priority to the configs we load last. We don't try and merge the contents of individual blocks.